### DISCUSSION TITLE
Update godot with ver

### DISCUSSION THREAD

**Discussion by st4rl1ght1337** [codemagic-ci-cd/codemagic-docs#1897] [Ideas]
Please change tutorial for Godot to have in represented that the user needs to set version under Godot_Version: {}. Additionally, the .yaml files for android, Windows and web should be provided given complexities in the Windows godot build process.

### Idea

```Text
https://blog.codemagic.io/godot-games-cicd/
```

**Comment by martinjeret** [comment]
Hey @st4rl1ght1337 would you be interested in helping us update the article?

**Reply by st4rl1ght1337** [reply]
Yes, working on it now. Also, is it possible to save the godot environment in some sort of persistent image / memory? The templates download is a bit heavy and would be better to cache it

**Reply by martinjeret** [reply]
re: godot environment - @Ycoi might have a better answer for this, but for Unity builds we recommend using some storage for static files, otherwise you can also use our caching: https://docs.codemagic.io/knowledge-codemagic/caching/

regarding the article update - please send me an email to martin@nevercode.io

Also thank you so much for volunteering to help and pointing out the issue!

**Reply by st4rl1ght1337** [reply]
Sure np, I'll email you the updated text when it's ready

**Reply by st4rl1ght1337** [reply]
One issue with caching, not sure about the right way to go about this. The dir that needs to be cached is /Users/builder/Library/Application Support/Godot/templates, but that directory won't be available until after the most recent version of Godot is installed, and I think it is not ideal to cache the version of Godot due to regular updates.

One idea that I have is to download the templates into a dir ~/templates and copy them to the /Users/.../templates, but if the volume is not on the same physical media it could take a long time. Also, templates are also updated frequently

**Reply by martinjeret** [reply]
Could you open a separate ticket with this question and tag @Ycoi who is our Product Manager. It is not the first time we have had questions similar to this (not necessarily Godot).

**Reply by Ycoi** [reply]
@st4rl1ght1337 Another option instead of caching could be to use Amozon S3 bucket or some other cloud storage for caching the files. The additional benefit in this approach is that you’re not limited by the 1GB cache limit of the Codemagic’s internal caching system. See [here](https://docs.codemagic.io/yaml-quick-start/building-a-unity-app/#using-your-own-external-storage-system) for more details on how to set it up (that’s for Unity but the approach should be applicable to Godot projects by changing the folders you want to cache)

As Martin said, if you have more ideas or feature requests about the caching or other topics, please open a separate ticket for those. Let’s keep this one for tracking when the docs are updated about the requirement for setting the Godot version.

**Comment by martinjeret** [comment]
I'll leave this issue open. Waiting on volunteer to contribute towards resolving this docs

**Comment by mikhail-tokarev** [comment]
While I’m closing this discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime
